### PR TITLE
Added custom globalException handler and validation for CreateBookReq…

### PR DESCRIPTION
**Pull request by dima-semeniuk**
…uestDto class

### CODE REVIEW

**Review by RostyslavKuzmych — APPROVED** [review]
Everything is nice!

File: src/main/java/mate/academy/webapp/exception/CustomGlobalExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        body.put("status", HttpStatus.BAD_REQUEST);
+        List<String> errors = ex.getBindingResult().getAllErrors().stream()
+                .map(this::getErrorMessage)
+                .toList();
+        body.put("errors", errors);
+        return new ResponseEntity<>(body, headers, status);
+    }
+
+    @ExceptionHandler(EntityNotFoundException.class)
+    protected ResponseEntity<Object> handleCustomsException(Exception ex) {
```
Comment (bhdnchui) [inline src/main/java/mate/academy/webapp/exception/CustomGlobalExceptionHandler.java:39]:
```suggestion
    protected ResponseEntity<Object> handleEntityNotFoundException(Exception ex) {
```

**Review by bhdnchui — CHANGES_REQUESTED** [review]
Just a few minor points to consider.

File: src/main/java/mate/academy/webapp/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -1,13 +1,20 @@
 package mate.academy.webapp.dto;
 
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Positive;
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class CreateBookRequestDto {
+    @NotNull(message = "can't be null.")
```
Comment (d1sam) [inline src/main/java/mate/academy/webapp/dto/CreateBookRequestDto.java:10]:
```suggestion
    @NotNull(message = "Title can't be null.")
```

File: src/main/java/mate/academy/webapp/dto/CreateBookRequestDto.java [hunk 3]
```diff
@@ -1,13 +1,20 @@
...
     private String title;
+    @NotNull(message = "can't be null.")
```
Comment (d1sam) [inline src/main/java/mate/academy/webapp/dto/CreateBookRequestDto.java:12]:
```suggestion
    @NotNull(message = "Author can't be null.")
```

File: src/main/java/mate/academy/webapp/dto/CreateBookRequestDto.java [hunk 4]
```diff
@@ -1,13 +1,20 @@
...
     private String author;
+    @NotNull(message = "can't be null.")
```
Comment (d1sam) [inline src/main/java/mate/academy/webapp/dto/CreateBookRequestDto.java:14]:
```suggestion
    @NotNull(message = "ISBN can't be null.")
```

File: src/main/java/mate/academy/webapp/dto/CreateBookRequestDto.java [hunk 5]
```diff
@@ -1,13 +1,20 @@
...
     private String isbn;
+    @NotNull(message = "can't be null.")
```
Comment (d1sam) [inline src/main/java/mate/academy/webapp/dto/CreateBookRequestDto.java:16]:
```suggestion
    @NotNull(message = "Price can't be null.")
```

File: src/main/java/mate/academy/webapp/exception/CustomGlobalExceptionHandler.java
Comment (d1sam) [inline src/main/java/mate/academy/webapp/exception/CustomGlobalExceptionHandler.java:1]: avoid string literals

**Review by d1sam — CHANGES_REQUESTED** [review]
Good job, but some changes should be done, please look at my comments.

**Review by kshuryhin — APPROVED** [review]
Looks good

**Pull request merged**